gapis/server/server.go

```diff
@@ -337,6 +338,35 @@ func (s *server) DCECapture(ctx context.Context, p *path.Capture, requested []*p
 	return trimmed, nil
 }
 
+func (s *server) SplitCapture(ctx context.Context, rng *path.Commands) (*path.Capture, error) {
+	ctx = log.Enter(ctx, "SplitCapture")
+	c, err := capture.ResolveGraphicsFromPath(ctx, rng.Capture)
+	if err != nil {
+		return nil, err
+	}
+
+	var from, to int
+	if len(rng.From) > 0 {
+		from = int(rng.From[0])
+	}
+	if len(rng.To) > 0 {
+		to = int(rng.To[0])
+	}
+
+	if from >= len(c.Commands) || to > len(c.Commands) || (to != 0 && to <= from) {
+		return nil, fmt.Errorf("Invalid range: [%d, %d) not in [%d, %d)", from, to, 0, len(c.Commands))
+	} else if to == 0 {
+		to = len(c.Commands)
+	}
+
+	name := fmt.Sprintf("%s [%d,%d)", c.Name(), from, to)
+	if gc, err := capture.NewGraphicsCapture(ctx, arena.New(), name, c.Header, c.InitialState, c.Commands[from:to]); err != nil {
```

Comment:
I think this will generate you a VERY unstable capture. Since the state block is *not* in the correct state. (Commands may reference non-existent objects, or other horrible issues). So keep this in mind. (You may want to document this).

Reply:
Indeed. At first I just had it be a truncate kind of thing where only the end index could be given, but then it was simple/only made sense to also allow a start index. I was thinking that it might be useful to allow a "forward state" flag that will mutate the initial state to the given start command, but didn't need it myself, so didn't spend the time on it.